test/playlist.test.js

```diff
@@ -650,11 +650,100 @@ QUnit.test('safeLiveIndex accounts for liveEdgePadding in non-simple case', func
   );
 
   assert.equal(
-    Playlist.safeLiveIndex(playlist, 0), 6,
+    Playlist.safeLiveIndex(playlist, 0), 5,
     'returns 6 when liveEdgePadding is 0'
   );
 });
 
+QUnit.test('getHoldBack works as expected', function(assert) {
+  const media = {
+    endList: true,
+    targetDuration: 5,
+    partTargetDuration: 1.1,
+    serverControl: {
+      holdBack: 20,
+      partHoldBack: 2
+    },
+    segments: [
+      {duration: 4},
+      {duration: 3},
+      {duration: 4, parts: [
+        {duration: 1},
+        {duration: 1},
+        {duration: 1},
+        {duration: 0.5},
+        {duration: 0.5}
+      ]}
+    ]
+  };
+  const master = {
+    suggestedPresentationDelay: 10
+  };
+
+  assert.equal(
+    Playlist.getHoldBack(master, media),
+    0,
+    'returns 0 with endlist'
+  );
+
+  delete media.endList;
+  assert.equal(
+    Playlist.getHoldBack(master, media),
+    master.suggestedPresentationDelay,
+    'uses suggestedPresentationDelay'
+  );
+
+  delete master.suggestedPresentationDelay;
+  assert.equal(
+    Playlist.getHoldBack(master, media),
+    media.serverControl.partHoldBack,
+    'uses part hold back'
+  );
+
+  media.serverControl.partHoldBack = null;
+  assert.equal(
+    Playlist.getHoldBack(master, media),
+    media.partTargetDuration * 3,
+    'uses part target duration * 3'
+  );
+
+  media.partTargetDuration = null;
+  assert.equal(
+    Playlist.getHoldBack(master, media),
+    2,
+    'uses last three part durations'
+  );
+
```

Comment:
May want another test here with fewer than 3 parts available.

Reply:
in a real scenario the last 3 segments will always have >= 2 parts. as having only one part is the same as not using parts at all. I will update the test for that.